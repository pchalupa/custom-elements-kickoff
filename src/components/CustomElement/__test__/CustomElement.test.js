import '../index';

describe('custom-element', () => {
	const ELEMENT_TAG = 'custom-element';
	let element;

	beforeEach(() => {
		element = document.createElement(ELEMENT_TAG);
		document.body.appendChild(element);
	});

	afterEach(() => {
		document.body.removeChild(element);
	});

	it('should be registered', () => {
		expect.assertions(1);

		expect(customElements.get(ELEMENT_TAG)).toBeDefined();
	});

	it('renders correctly', () => {
		expect.assertions(2);

		expect(element.classList.contains('container')).toBeTruthy();
		expect(element).toMatchSnapshot();
	});
});
