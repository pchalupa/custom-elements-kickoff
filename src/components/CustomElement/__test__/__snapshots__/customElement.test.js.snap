// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom-element renders correctly 1`] = `
<custom-element
  class="container"
>
  <h1 />
</custom-element>
`;
